v56-team22-surgery-status-board/src/components/PatientStatusTable/PaginationController.tsx

```diff
@@ -0,0 +1,134 @@
+import  { useState } from 'react';
+import { type Table as ReactTable } from '@tanstack/react-table';
+import {
+  PaginationNext,
+  PaginationPrevious,
+} from '@/components/ui/pagination';
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from '@/components/ui/select';
+import { Button } from '@/components/ui/button';
+
+import {
+  FaAngleDoubleLeft,
+  FaAngleDoubleRight,
+  FaAngleLeft,
+  FaAngleRight,
+} from 'react-icons/fa';
+
+// type PaginationMode = 'manual' | 'auto' | 'paused';
+interface PaginationControllerProps<TData> {
+  table: ReactTable<TData>;
+}
+
+function PaginationController<TData>({
+  table,
+}: PaginationControllerProps<TData>) {
+  const [pageSize, setPageSize] = useState(
+    table.getState().pagination.pageSize
+  );
+
+  const totalPages = table.getPageCount();
+  const currentPage = table.getState().pagination.pageIndex;
+  const totalItems = table.getRowModel().rows.length;
+  const startIndex = currentPage * pageSize + 1;
+  const endIndex = Math.min((currentPage + 1) * pageSize, totalItems);
+
+  const handlePageSizeChange = (value: string) => {
+    const size = parseInt(value);
+    setPageSize(size);
+    table.setPageSize(size);
+  };
+
+  //TODO: Implement auto pagination mode
+
+  // Helper to determine disabled state for Previous/Next
+  const isPreviousDisabled = currentPage === 0;
+  const isNextDisabled = currentPage === totalPages - 1;
+
+  return (
+    <div className="flex flex-col items-center gap-4 py-4">
+      <div className="text-sm text-gray-600">
+        {totalItems > 0
```

Comment:
It seems like the `endIndex` is not being calculated correctly because of the `totalItems` calculation not reflecting all items in the table and only on the screen.  <img width="365" height="75" alt="Screenshot 2025-08-07 at 15 24 48" src="https://github.com/user-attachments/assets/821dfe93-9def-4431-b95c-0a67d3cfd9ea" />